airbrakes/hardware/imu.py

```diff
@@ -64,99 +70,117 @@ def __init__(self, port: str) -> None:
-        for data_point in packet.data():
-            # Extract the channel name of the data point.
-            channel = data_point.channelName()
-
-            # Check if the channel is relevant for the data packet, we check for quaternions
-            # separately because the IMU sends them over as a matrix, but we store each of them
-            # individually as fields.
-            if hasattr(imu_data_packet, channel) or "Quaternion" in channel:
-                # Process and set the data point value in the data packet.
-                IMU._process_data_point(data_point, channel, imu_data_packet)
-
-    def _fetch_data_loop(self, port: str) -> None:
+    def _fetch_data_loop(self, port: str) -> None:  # pragma: no cover
         """
         Continuously fetch data packets from the IMU and process them.
         :param port: The serial port to connect to the IMU.
         """
+        # Set the process priority really high, as we want to get the data from the IMU as fast as
+        # possible:
+        set_process_priority(-15)
+
         # Connect to the IMU and initialize the node used for getting data packets
         connection = mscl.Connection.Serial(port)
         node = mscl.InertialNode(connection)
+        packet: mscl.MipDataPacket
+        data_point: mscl.MipDataPoint
 
         while self.is_running:
             # Retrieve data packets from the IMU.
             packets: mscl.MipDataPackets = node.getDataPackets(timeout=10)
 
-            packet: mscl.MipDataPacket
+            self._fetched_imu_packets.value = len(packets)
+
+            messages = []
             for packet in packets:
-                # Initialize the appropriate data packet.
-                imu_data_packet = IMU._initialize_packet(packet)
-                if imu_data_packet is None:
-                    # Skip unrecognized packets.
+                # Extract the timestamp from the packet.
+                timestamp = packet.collectedTimestamp().nanoseconds()
+
+                # Initialize packet with the timestamp, determines if the packet is raw or estimated
+                if packet.descriptorSet() == ESTIMATED_DESCRIPTOR_SET:
+                    imu_data_packet = EstimatedDataPacket(timestamp)
+                elif packet.descriptorSet() == RAW_DESCRIPTOR_SET:
+                    imu_data_packet = RawDataPacket(timestamp)
+                else:
                     continue
 
-                # Process the packet's data and populate the data packet object.
-                IMU._process_packet_data(packet, imu_data_packet)
-
-                # Add the processor data packet to the shared queue.
-                self._data_queue.put(imu_data_packet)
+                # Iterate through each data point in the packet.
+                for data_point in packet.data():
+                    # Extract the channel name of the data point.
+                    match data_point.field(), data_point.qualifier():
+                        case 32775, 1:
```

Comment:
todo: test if this is faster than just using `if-else`? If it isn't, write a comment saying by how much it is slower by.

Reply:
so if-elif-else statements are faster. Benchmarked this:

- Current implementation with `match` statements: 7546 ns.
- With if-elif-else: 4188 ns. 

I ran it for 15 seconds, so it should be a good representation of the performance difference